tests/core/basic20/task.yaml

```diff
@@ -26,20 +26,22 @@ execute: |
     echo "Ensure passwd/group is available for snaps"
     test-snapd-sh-core20.sh -c 'cat /var/lib/extrausers/passwd' | MATCH test
 
-    #shellcheck source=tests/lib/names.sh
-    . "$TESTSLIB"/names.sh
+    if [[ "$SPREAD_SYSTEM" != ubuntu-core-20-arm-* ]]; then
+        #shellcheck source=tests/lib/names.sh
+        . "$TESTSLIB"/names.sh
 
-    echo "Ensure extracted kernel.efi exists"
-    test -e /boot/grub/"$kernel_name"*/kernel.efi
-    test -e /boot/grub/kernel.efi
+        echo "Ensure extracted kernel.efi exists"
+        test -e /boot/grub/"$kernel_name"*/kernel.efi
```

Comment:
instead of not doing any check for the kernel on arm devices, we should instead check for the extracted kernel assets @ /run/mnt/ubuntu-seed/systems/<label>/kernel/"$kernel_name"*/{kernel,initrd}.img like we do in the test I just updated:

https://github.com/snapcore/snapd/pull/9646/files#diff-13cee10370ef27346da1ad984f529abd35b91d8eb34dce331a1a3c4c88a06f7aR10

Reply:
perfect